docusaurus/docs/packages/pkg/relayer/relayminer_config.md

```diff
@@ -0,0 +1,379 @@
+---
+title: RelayMiner config
+sidebar_position: 1
+---
+
+## `relayer/config/relayminer_configs_reader`
+
+_This document describes the configuration options available through the
+`relayminer_config.yaml` file. It drives how the `RelayMiner` is setup in terms
+of Pocket network connectivity, the servers it starts, which domains it accepts
+queries from and which services it forwards requests to._
+
+- [Usage](#usage)
+- [Structure](#structure)
+- [Top level options](#top-level-options)
+  - [`signing_key_name`](#signing_key_name)
+  - [`smt_store_path`](#smt_store_path)
+- [Pocket node connectivity](#pocket-node-connectivity)
+  - [`query_node_rpc_url`](#query_node_rpc_url)
+  - [`query_node_grpc_url`](#query_node_grpc_url)
+  - [`tx_node_grpc_url`](#tx_node_grpc_url)
+- [RelayMiner proxies](#relayminer-proxies)
+  - [`name`](#name)
+  - [`type`](#type)
+  - [`host`](#host)
+- [Suppliers](#suppliers)
+  - [`name`](#name-1)
+  - [`type`](#type-1)
+  - [`service_config`](#service_config)
+    - [`url`](#url)
+    - [`authentication`](#authentication)
+    - [`headers`](#headers)
+  - [`hosts`](#hosts)
+  - [`proxy_names`](#proxy_names)
+- [Examples](#examples)
+  - [Proxy -\> Supplier referencing](#proxy---supplier-referencing)
+  - [Full config example](#full-config-example)
+
+# Usage
+
+The `RelayMiner` start command accepts a `--config` flag that points to a configuration
+`yaml` file that will be used to setup the `RelayMiner` instance.
+
+```bash
+poktrolld relayminer --config ./relayminer_config.yaml --keyring-backend test
+```
+
+# Structure
+
+The `RelayMiner` configuration file is a `yaml` file that contains `top level options`,
+`proxies` and `suppliers` sections.
+
+# Top level options
+
+```yaml
+signing_key_name: <string>
+smt_store_path: <string>
+```
+## `signing_key_name`
+ _`Required`_
+
+The name of the key that will be used to sign transactions, derive the public key
+and the corresponding address. This key name must be present in the keyring that is used
+to start the `RelayMiner` instance.
+
+## `smt_store_path`
+_`Required`_
+
+The relative or absolute path to the directory where the `RelayMiner` will store
+the `SparseMerkleTree` data on disk. This directory is used to persist the `SMT`
+in a BadgerDB KV store data files.
+
+# Pocket node connectivity
+
+```yaml
+pocket_node:
+  query_node_rpc_url: tcp://<host>
```

Comment:
Should we add a `:port` as well?

Reply:
`<host>` usually refers to `<hostname>:<port>`, we can change it to that if it's clearer.

Reply:
Good point. 

Fwiw, I think this change is only a net positive.